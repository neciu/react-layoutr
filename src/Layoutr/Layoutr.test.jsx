/* eslint-env jest */

import React from 'react';
import { shallow, mount } from 'enzyme';

import Layoutr from './Layoutr';

test('should be a div', () => {
  const wrapper = shallow(<Layoutr />);

  const type = wrapper.type();

  expect(type).toBe('div');
});

test('should match snapshot', () => {
  const wrapper = mount(<Layoutr />);

  expect(wrapper).toMatchSnapshot();
});

test('should have display flex', () => {
  const tree = shallow(<Layoutr />);

  expect(tree).toHaveStyleRule('display', 'flex');
});

test('should handle passed width', () => {
  const integerWidthWrapper = shallow(<Layoutr width={100} />);

  expect(integerWidthWrapper).toHaveStyleRule('width', '100px');
});
