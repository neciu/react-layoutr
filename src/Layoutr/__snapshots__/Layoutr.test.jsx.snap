// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

<styled.div>
  <div
    className="c0"
  />
</styled.div>
`;
